bundles/org.openhab.binding.panamaxfurman/src/main/java/org/openhab/binding/panamaxfurman/internal/protocol/PowerConditionerChannel.java

```
@@ -0,0 +1,119 @@
+ *
+ * @author Dave Badia - Initial contribution
+ */
+@NonNullByDefault
+public enum PowerConditionerChannel {
+    POWER("power", OnOffType.class, true),
+    ID("id", null, false),;
+
+    private final static Logger logger = LoggerFactory.getLogger(PowerConditionerChannel.class);
+
+    private final static Map<String, PowerConditionerChannel> CHANNEL_STRING_TO_ENUM_TABLE = new ConcurrentHashMap<>();
+
+    private final String name;
+    private final @Nullable Class<? extends State> stateClass;
+    private final boolean outletSpecific;
+
+    private PowerConditionerChannel(String name, @Nullable Class<? extends State> stateClass, boolean outletSpecific) {
+        this.name = name;
+        this.stateClass = stateClass;
+        this.outletSpecific = outletSpecific;
+    }
+
+    public String getChannelName(@Nullable Integer outletNumber) {
+        if (outletNumber == null) {
+            return getName();
+        }
+        if (isOutletSpecific()) {
+            return PanamaxFurmanAbstractHandler.getChannelUID(name, outletNumber);
+        } else {
+            logger.warn("{} was called with outlet #{} but is not outlet specific. ()", this, outletNumber);
+            return name;
+        }
+    }
+
+    public String getName() {
+        if (isOutletSpecific()) {
+            throw new IllegalStateException(this.name() + " was called without an outlet # but is outlet specific");
+        }
+        return name;
+    }
+
+    public @Nullable Class<? extends State> getStateClass() {
+        return stateClass;
+    }
+
+    public @Nullable State buildState(String stateValueParam) {
+        String stateValue = stateValueParam.trim();
+        if (getStateClass() == null) {
+            return null;
+        } else if (OnOffType.class.equals(getStateClass())) {
+            return OnOffType.from(stateValue);
+        } else {
+            logger.warn("Don't know how to map state class {}, returning null", stateClass);
```

```suggestion
            logger.warn("Don't know how to map state class {}, returning null", stateClass);
```

There are more warn level, are you sure all these needs to be warn instead of debug ? Is there anything the user can do about it or should act (like raise an issue for unsupported device or something liek that)

---

Yes, the user should definitely raise an issue if this particular warning occurs.  I'll double check the other warn to see if I can reduce the log level

---

Her eyou could also help the user to tell him what to do:
"Can't map state class {}, returning null. Please raise a Github issue"

---

Good point, updated